src/sqlfluff/testing/rules.py

```diff
@@ -87,11 +87,15 @@ def assert_rule_pass_in_sql(code, sql, configs=None):
     # Configs allows overrides if we want to use them.
     cfg = FluffConfig(configs=configs)
     r = get_rule_from_set(code, config=cfg)
-    parsed = Linter(config=cfg).parse_string(sql)
+    linter = Linter(config=cfg)
```

Comment:
Changed this code so we can get and pass along the `TemplatedFile` instance to the `crawl()` function.

It turns out we don't need to change the public interface to `Linter` at all. The existing API works, although some of these secondary functions like `render_string()` have a poorly designed API (seems like I should be able to pass it just the `sql` string, but it requires those 3 other parameters).